Solutions/Branwen/GUI.cs

```diff
@@ -0,0 +1,166 @@
+        /// </summary>
+        /// <param name="parent"></param>
+        /// <returns></returns>
+        private List<FileInfo> RunInventory(DirectoryInfo parent)
+        {
+            List<FileInfo> toReturn = new List<FileInfo>();
+            foreach (DirectoryInfo directory in parent.GetDirectories())
+            {
+                toReturn.AddRange(RunInventory(directory));
+            }
+            toReturn.AddRange(parent.GetFiles());
+            return toReturn;
+        }
+
+        /// <summary>
+        /// Writes all files in the Directory to the Workbook
+        /// </summary>
+        /// <param name="files"></param>
+        /// <param name="writer"></param>
+        private void WriteDirectoryToWorksheet(List<FileInfo> files, SheetData sheetData)
+        {
+            //Write Worksheet Header
+            WriteHeader(sheetData);
+
+            for (int i = 0; i < files.Count; i++)
+            {
+                Row row = new Row();
+                Cell cell = new Cell();
+                
+                //File Name
+                cell.CellValue = new CellValue(Path.GetFileNameWithoutExtension(files[i].Name));
+                cell.DataType = CellValues.String;
+                row.Append(cell);
+
+                //File Extension
+                cell = new Cell();
+                cell.CellValue = new CellValue(Path.GetExtension(files[i].Name));
+                cell.DataType = CellValues.String;
+                row.Append(cell);
+
+                //Filesize
+                cell = new Cell();
+                cell.CellValue = new CellValue((files[i].Length / 1048576).ToString());
+                cell.DataType = CellValues.Number;
+                row.Append(cell);
+
+                //write all the pathNames to the spreadsheet
+                List<string> path = files[i].DirectoryName.Split('\\').ToList(); ;
+                foreach (string pathName in path)
+                {
+                    cell = new Cell();
+                    cell.CellValue = new CellValue(pathName);
+                    cell.DataType = CellValues.String;
+                    row.Append(cell);
+                }
+
+                sheetData.Append(row);
+                
+                fileCount++;
+            }
+        }
+
+        /// <summary>
+        /// Write the standard header to the Worksheet
+        /// </summary>
+        /// <param name="worksheet"></param>
+        private void WriteHeader(SheetData sheetData)
+        {
+            Row row = new Row();
+            List<string> headers = new List<string>() { "File Name", "File-Type", "Size(In MB)", "Folder1", "Folder2", "Folder3", "Folder4", "Folder5", "Folder6", "Folder7" };
+            foreach(string header in headers)
+            {
+                Cell someCell = new Cell();
+                someCell.CellValue = new CellValue(header);
+                someCell.DataType = CellValues.String;

```

Comment:
remove this from the loop too, you are just setting it to the same value every iteration